docs/use_cases/fine_grained_authorization/intro.md

```diff
@@ -0,0 +1,180 @@
+documents, bank accounts, shopping carts, etc. The edges in this graph capture the **relationships**
+between these objects, such as `parent`, `owner`, `member`, etc. The specific types of objects and relationships
+vary depending on the application’s requirements, allowing developers to implement application-specific
+authorization policies.
+
+#### Example: shared file manager
+
+We will use a simple shared file management service as a running example throughout this guide.
+This application works with three types of objects:
+
+- **Users** represent individual users of the service.
+- **User groups** group users with similar permissions.
+- **Files and folders**. For simplicity, we model files and folders using a single object type. A folder
+  is simply a file that has one or more children.
+
+Objects are connected by the following relationships:
+- `member` - a relationship between a user and a group.
+- `parent` - a relationship between files that defines the folder hierarchy.
+- `editor` - a relationship between a group and a file that gives the group the permission to read or write the file.
+- `viewer` - a relationship between a group and a file that gives the group the permission to read the file.
+
+We use `relationship(object1, object2)` notation for relationships, e.g., `member(alice, engineering)`, `parent(folder1, file1)`,
+`editor(admins, folder1)`, etc.
+
+Here is an example object graph that shows several users, groups, and files.  Note that objects can have
+**attributes**, e.g., `is_banned`.
+
+<a id="example-object-graph"></a>
+![Object graph](./object-graph.png)
+
+### Rules
+
+An FGA policy consists of rules that derive new relationships from the object graph.  A rule consists
+of **prerequisites** that must hold for the rule to fire and the **derived clause** that holds if the prerequisites
+are satisfied. The language of valid rules depends on a specific FGA implementation. In this tutorial we
+consider two most common types of rules:
+
+1. Rules with one prerequisite:
+  ```
+  relationship1(object1, object2) and condition(object1, object2) -> relationship2(object1, object2)
+  ```
+  where `condition` is an arbitrary predicate over atributes of `object1` and `object2`.  This rule states
+  that if there exists relationship `relationship1` between `object1` and `object2`, which satisfy `condition`,
+  then `object1` and `object2` are also related by `relationship2`.
+
+2. Rules with two prerequisites:
+  ```
+  relationship1(object1, object2) and relationship2(object2, object3) and condition(object1, object3) -> relationship3(object1, object3)
+  ```
+
+More complex rules, e.g., rules with more than two prerequisites or rules with arbitrary combinations of `and` and `or`
+operators, can be decomposed into these two basic types. Some FGA implementations allow negative prerequisites of the
+form `not relationship1(object1, object2)`. Such rules can be expressed in Feldera, but we leave them out for simplicity.
```

Comment:
leave them for future discussion?

Reply:
what's wrong with "for simplicity"? I basically don't want to overload this already fairly complex example.

Reply:
I just said that in the future you may want to discuss them.